### PR TITLE
add updated linked list

### PR DESCRIPTION
**Pull request by NasreenParween**
#### Checklist

- [ ] Added description of change
- [ ] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [ ] Added tests and example, test must pass
- [ ] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [ ] Relevant documentation/comments is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [ ] Search previous suggestions before making a new one, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

**Review by mishraabhinn — COMMENTED** [review]
Plz make one pr per file 
Contributing guidelines

**Comment by NasreenParween** [comment]
@mishraabhinn Added the required changes

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Hey there! Thank you for submitting a PR! 😄 Sadly, this does not meet our requirements. Check out our [repository standards](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md). Please read them **carefully** and **follow them**. **Let us know if you need any help here or on our [Discord server](https://the-algorithms.com/discord).** 🙂

**Comment by DxLRate** [comment]
how can I contribute?

**Comment by mishraabhinn** [comment]
> how can I contribute?

[Contributing Guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md)

**Comment by mishraabhinn** [comment]
@Panquesito7 and @NasreenParween I really feel this is already being implemented. I wish to close this pr. Thanks for your contribution.  Please, join discord and gitter to remain more and more connected to the algorithms. https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/data_structures/linked_list.cpp

**Comment by Panquesito7** [comment]
> @Panquesito7 and @NasreenParween I really feel this is already being implemented. I wish to close this pr. Thanks for your contribution. Please, join discord and gitter to remain more and more connected to the algorithms. https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/data_structures/linked_list.cpp

Hmm, I'm not sure if we should keep it; after all, it's another implementation and I think we should accept it as long as it follows our contributing guidelines, of course. What do you think? 🙂
CC: @tjgurwara99.

File: data_structures/singly_linked_list2.cpp [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+template<class t>
+class linkedlist{
+    public:
+    node<t> *start;
+    // Default constructor
+    linkedlist(){
+        start=NULL;
+    }
+    node<t>  *createnewnode(t val){
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:29]: Add empty line and function description.

File: data_structures/singly_linked_list2.cpp [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        node<t> *newnode;
+        newnode=new node<t>();
+        newnode->info=val;
+        newnode->next=NULL;
+
+        return (newnode);
+    }
+    void createnewlist(node<t> *newnode){
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:37]: Add empty line and function description.

File: data_structures/singly_linked_list2.cpp [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+            node<t> *temp;
+            temp=start;
+            newnode->next=temp;
+            start=newnode;
+    }
+    // Function to insert a node at the end of the linked list.
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:43]: Add empty line

File: data_structures/singly_linked_list2.cpp [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    void addatend(node<t> *newnode){
+         node<t>* temp;
+ if(start==NULL)
+ {
+ start=newnode;
+ return;
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:49]: Indentation

File: data_structures/singly_linked_list2.cpp [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+ }
+ for(temp=start; temp->next!=NULL; temp=temp->next)
+ {
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:52]: while loop can be better

File: data_structures/singly_linked_list2.cpp [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+ }
+ temp->next=newnode;
+}
+// Function to insert a after the node the linked list.
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:56]: Add empty line

File: data_structures/singly_linked_list2.cpp [hunk 7]
```diff
@@ -0,0 +1,313 @@
...
+            node<t> *temp;
+            for(temp=start;temp!=NULL;temp=temp->next){
+                if(temp->info==after){
+                    newnode->next=temp->next;
+                    temp->next=newnode;
+                }
+            }
+    }
+
+    void addbefore(node<t> *newnode){
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:70]: Add description. Use camelCase for function names

File: data_structures/singly_linked_list2.cpp [hunk 8]
```diff
@@ -0,0 +1,313 @@
...
+                if(temp->next->info==before){
+                    newnode->next=temp->next;
+                    temp->next=newnode;
+                    break;
+                }
+            }
+        }
+    }
+
+    t deletespecific(t i){
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:92]: Add function description.

File: data_structures/singly_linked_list2.cpp [hunk 9]
```diff
@@ -0,0 +1,313 @@
...
+            node<t> *temp;
+            int del;
+            int c;
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:95]: Initialize variables with default values. Multiple places found.

File: data_structures/singly_linked_list2.cpp [hunk 10]
```diff
@@ -0,0 +1,313 @@
...
+                        del=p->info;
+                        temp->next=temp->next->next;
+                        delete p;
+                    }
+                }
+            }
+                return del;
+    }
+    // Function to delete the node at given position
+
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:124]: Remove empty line and add before description

File: data_structures/singly_linked_list2.cpp [hunk 11]
```diff
@@ -0,0 +1,313 @@
...
+                }
+            }
+        }
+
+    }
+
+    int count(){
+        int ct=0;
+        node<t> *temp;
+        for(temp=start;temp!=NULL;temp=temp->next){
```
Comment (krishnacx) [inline data_structures/singly_linked_list2.cpp:151]: Use while

File: data_structures/singly_linked_list2.cpp [hunk 12]
```diff
@@ -0,0 +1,313 @@
...
+ * @ linked list using template
+ * @brief The linked list stores data in sequential storage, like arrays. Though the data are stored sequentially,
+   the memory locations are not contiguous. Unlike an array, the linked list can store data of different data types.
+
+ * @author Nasreen parween (https://github.com/NasreenParween)
+ * @see singly_linked_list.cpp, stack_using_linked_list.cpp, circular_linked_list.cpp ,doubly_linked_listn.cpp
+ */
+
+#include<iostream>
+using namespace std;
```
Comment (tjgurwara99) [inline data_structures/singly_linked_list2.cpp:11]: You've used this but I suggest moving your implementation inside a separate `namespace` under the `data_structures` namespace. For reference, take a look [here](https://github.com/TheAlgorithms/C-Plus-Plus/blob/1371b5390abc98ab127c5a9057cbeff6e2afbbfb/data_structures/linked_list.cpp#L21-L31).

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]
Once you have moved it into the desired namespace, I would review this again properly.

**Comment by tjgurwara99** [comment]
> Hmm, I'm not sure if we should keep it; after all, it's another implementation and I think we should accept it as long as it follows our contributing guidelines, of course. What do you think? 🙂
> CC: @tjgurwara99.

I'm a bit confused, by different implementation do you mean that the nitty gritty details are different? I don't think there is much difference to it than the one we already have if you're considering the difference in logic but if its just the implementation then I guess we can accept it.

**Pull request closed without merging**